templates/stamp/utilities.sh

```diff
@@ -618,8 +618,13 @@ sync_repo()
     else
         pushd $repo_path
 
-        sudo git fetch --all --tags --prune
-        exit_on_error "Failed syncing repository $repo_url to $repo_path"
+        if ! sudo git pull --all --tags --prune ; then
```

Comment:
What's being addressed here? git pull incorporates a git fetch + git merge. Why is the below still necessary?

Reply:
yes - therefore `git pull` fails when `git merge` is impossible.
specifically, `git merge` fails when there is no corresponding branch in the upstream to use. there are a LOT of cases when this can happen, but one case I've run into with this function is when the local folder is sync'd to a tag (so a specific commit not necessarily associated with a branch) before the function is called.